Hackerspaces/PrefPaneTableViewController.swift

```diff
@@ -15,18 +15,40 @@ class PrefPaneTableViewController: UITableViewController {
             toggle.isOn = SharedData.isInDebugMode()
         }
     }
+
     @IBAction func toggleDebug(_ sender: UISwitch) {
 
         SharedData.toggleDebugMode()
         toggle.isOn = SharedData.isInDebugMode()
         if toggle.isOn {
-            let alert = UIAlertController(title: "Debug mode enabled", message: "debug mode allows you to access error details when a hackerspace is unreachable", preferredStyle: .alert)
-            alert.addAction(UIAlertAction(title: "OK", style: .default, handler: nil))
-            present(alert, animated: true, completion: nil)
+            let alert = UIAlertController(title: "Advanced mode enabled", message: "Advanced mode displays more advanced features useful for hackerspace API developers", preferredStyle: .alert)
+            alert.addAction(UIAlertAction(title: "OK", style: .default, handler: constArg(tableView.reloadData)))
+            present(alert, animated: true, completion:         nil)
+        } else {
+            tableView.reloadData()
         }
     }
     
-    override func viewDidLoad() {
-        super.viewDidLoad()
+    override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        if section == 0 {
+            return toggle.isOn ? 3 : 2
+        } else {
+            return 5
+        }
+    }
+
+    override func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        print("selected indexpath \(indexPath)")
+        if indexPath.section == 0, indexPath.row == 2 {
+            askConfirmation(for: SharedData.deleteAllDebug)
```

Comment:
This should be renamed and repurposed as a generic alert factory.
Making a function that is only called once only creates a needless level of indirection.